Merge latest developments from GEOMAR into release

seb-wahl opened the pull request:
Hi
this will merge mostly FOCIOIFS related developments. 
Still contains the bug described by @joakimkjellsson in #1137

seb-wahl committed:
fix previously introduced bug: use correct archfile with AGRIF configurations

seb-wahl committed:
added AGRIF configuration without AOW tracer

seb-wahl committed:
added missint nemo-ORCA05_LIM2_FOCI_AGRIF+echam-6.3.05p2-foci folder

seb-wahl committed:
fix leap year settings

Commit:
started adding things for Tco399-ORCA12. Also add ln_single_prec_send to runscript

Commit:
Merge branch 'geomar_dev' of https://github.com/esm-tools/esm_tools into geomar_dev

Commit:
oifs annual means was not working. Seems ok now.

seb-wahl committed:
bugfixes in monitoring for nemo

Commit:
added runscript for spinup-1950 for foci-openifs

Commit:
fix small conflict

seb-wahl committed:
Merge branch 'geomar_dev' of https://github.com/esm-tools/esm_tools into geomar_dev

seb-wahl committed:
added mocsigma and 26.5N AMOC MAX

Commit:
clean up focioifs and fix runoff for AGRIF

Commit:
FOCI-OpenIFS-VIKING10 works on glogin with proper runoff remapping

Commit:
add settings for ORION10X

seb-wahl committed:
added a few foci runscript examples

seb-wahl committed:
Merge branch 'release' into geomar_dev

seb-wahl committed:
added correct nest_refinement for WG10v4

seb-wahl committed:
first draft of eORCA025.z75.SI3, still crashes with diamond shape patterns of NaN
during the first time step

seb-wahl committed:
fixed two bugs in nemo postprocessing (date math)

seb-wahl committed:
added nemo4.2 configuration that ran 3 years using JRA-55 forcing and the v1.0.5 Levitus initial data

seb-wahl committed:
fixed date bugs in nemo_postprocessing.sh

seb-wahl committed:
added test queue on HLRN4 machines

seb-wahl committed:
Merge pull request #908 from esm-tools/feature/add-eORCA025-standalone

Feature/add eorca025 standalone

seb-wahl committed:
remove set -vx

seb-wahl committed:
focioifs-3.0 compiles, won't run as now NEMO4.2 specific adjustments have been made
updated nemo4 example standalone runscripts.

seb-wahl committed:
add _nemo4 to compile settings for glogin
uncomment compiler_mpi in blogin.yaml again

seb-wahl committed:
fix a typo in blogin.yaml

seb-wahl committed:
add TCO95_eORCA025 switch

Commit:
change xml files for NEMO4, and add new OpenIFS dev branch for 3.0

Commit:
Merge branch 'feature/add-focioifs-3.0' of https://github.com/esm-tools/esm_tools into feature/add-focioifs-3.0

Commit:
add gcc 9.3.0 to main module for compiling FOCI-OpenIFS on glogin

Commit:
new script for generating OpenIFS initial conditions from GRIB output. If all necessary fields are written in output, then this script can be used

Commit:
bug fix for oifs processing when running annual restarts

Commit:
add LNEMOLIMCUR to run with relative winds in FOCI-OIFS

seb-wahl committed:
fixed modules for NEMO4
first modifications for NEMO4 coupled to openIFS

seb-wahl committed:
focioifs-3.0 technically up and running (piCtl)

seb-wahl committed:
Merge pull request #928 from esm-tools/feature/add-focioifs-3.0

Feature/add focioifs 3.0

seb-wahl committed:
Merge pull request #929 from esm-tools/release

merge the latest release into geomar_dev

Commit:
add abrupt 4xCO2 scenario to oifs.yaml. Also add runscript to branch off historical and 4xCO2 experiments from FOCI-OpenIFS control

Commit:
added eoptmized nodes for glogin

Commit:
fix for branch-off experiments with eternal restarts in OpenIFS. Now set ini_pseudo_initial_date in runscript for restart. Also, fix for 1pctCO2 which was not working before.

Commit:
openifs runs on NESH, but post fails with git exec error

Commit:
oifs-43r3-v2 is now based on branch foci211. Also fixed postprocessing for AMIP runs

Commit:
new xml files

Commit:
add orbital parameters to climMEMODEL runscript

seb-wahl committed:
Merge pull request #983 from esm-tools/release

update geomar_dev with latest release

Commit:
module load git git-lfs on nesh

Commit:
Merge branch 'geomar_dev' of https://github.com/esm-tools/esm_tools into geomar_dev

Commit:
set ini_pseudo_initial_date in case not already set

Commit:
add L75 option. Also add Calving mask for AGRIF, e.g. ORION10

Commit:
add Intel 2022 / IMPI 2021 as compile option on glogin. The older, Intel 2019, stopped working

Commit:
add settings for calving on AGRIF, and also Tco319-ORCA12 config. Also set default compiler on glogin to Intel 2022/IMPI 2021

Commit:
runscript to run Tco319-ORCA12 on glogin. Works, but slower than Tco95 version. Likely due to high I/O from XIOS

Commit:
Merge branch 'geomar_dev' of https://github.com/esm-tools/esm_tools into geomar_dev

Commit:
update IO_LIB_ROOT for Intel 2022 compilers on glogin

Commit:
New Intel compilers (2022 and beyond) do not accept -mkl=sequential anymore. Replace by -qmkl=sequential, which should work on any new-ish versions.

Commit:
fix so that clean command also removed BLD and arch_nemo.fcm for NEMO.

Commit:
fixed so that month 08 is no longer interpreted as an octal number. Should only impact names of postprocessed OpenIFS output.

Commit:
add OpenIFS output on theta levels to oifs_postprocessing.sh

joakimkjellsson committed:
switch domain_cfg file for NEMO4. The old one was bad

joakimkjellsson committed:
add gcc so that XIOS compiles properly

joakimkjellsson committed:
bugfix so that nemo post script works for monthly restarts properly. 08 was being interpreted as octal numbers...

joakimkjellsson committed:
better default nproc settings for FOCI-OpenIFS 4.0

joakimkjellsson committed:
make new version: FOCI-OpenIFS 2.2 with new switches to use 1st order conserv remapping with oasis

seb-wahl committed:
first draft of fs_oasismct4
cleanup still required

joakimkjellsson committed:
introduce FOCI-OpenIFS v2.2 using OASIS3-MCT master branch directly from CERFACS

joakimkjellsson committed:
add clean command for oasis mct5

joakimkjellsson committed:
add file_def_agrif for agrif-3.0

joakimkjellsson committed:
we can now choose conserv (1st order) or gauswgt for atm-agrif remapping. conserv does not work though

joakimkjellsson committed:
new way to activate load balancing tool (LUCIA) in OASIS MCT5

joakimkjellsson committed:
added FOCI-OpenIFS 3.0_busan. OpenIFS 43r3 + NEMO 4.2 (ORCA05 L46)

joakimkjellsson committed:
added FOCI-OpenIFS 2.2 with OASIS-MCT5. Works on glogin

joakimkjellsson committed:
add correct xml files for nemo4 (focioifs 3.0_busan)

joakimkjellsson committed:
Merge pull request #1101 from esm-tools/feature/oasis_from_cerfacs

Feature/oasis from cerfacs

joakimkjellsson committed:
add variable oasis_p to control if we set P 2 (for NEMO 3.6) or P 0 (for NEMO 4) in namcouple file. This controls if periodic points are included in the coupling fields or not

joakimkjellsson committed:
switch to OASIS MCT5 for Busan version of FOCI-OpenIFS

joakimkjellsson committed:
change MCT version also here

joakimkjellsson committed:
add bilinear and bicubic remapping methods for FOCI-OpenIFS. This means we handle everything as bilinear but use bicubic for wind stress

joakimkjellsson committed:
output contains z on surface, not hybrid level

joakimkjellsson committed:
updated runscript for focioifs3_busan

joakimkjellsson committed:
change leapyear to true for ORCA05 coupled config with NEMO4. Warning: If leapyear is false in NEMO coupled to OpenIFS, ESM-Tools and OASIS wont complain, but XIOS will throw strange errors after 3 years

seb-wahl committed:
Merge pull request #1110 from esm-tools/focioifs3_busan

Focioifs3 busan

joakimkjellsson committed:
switch domain_cfg for NEMO4-ORCA05 to exclude Caspian Sea

joakimkjellsson committed:
Switch NEMO4 version to NEMO 4.2.2 (ice drag bugfix)

joakimkjellsson committed:
switch to intel 2022 on glogin. seems fine and its good to be up to date

joakimkjellsson committed:
add NEMO4 output to postprocessing. diaptr2D etc

joakimkjellsson committed:
add ICMSH and ICMGG files to output files for OpenIFS

joakimkjellsson committed:
update runscripts for FOCI-OpenIFS

joakimkjellsson committed:
update runscripts for FOCI-OpenIFS

joakimkjellsson committed:
Merge pull request #1131 from esm-tools/focioifs3_busan

Focioifs3 busan

joakimkjellsson committed:
added regular_pl_zoom output for OpenIFS, which is pressure-level output but only for 850, 500 and 300 hPa. Useful to output high-frequency fields while keeping data size down

joakimkjellsson committed:
added runscripts for HighResMIP runs with FOCI-OpenIFS-VIKING10

seb-wahl committed:
added (I know it's the wrong branch)
- gauswght direct coupling for FOCI
- a few missing runscripts
- bugfix for JSBACH for hist runs

seb-wahl committed:
removed -l from default shell interpreter, caused problems on HLRN
fix search pattern for restart file naming convention used with FOCI

seb-wahl committed:
Merge pull request #1132 from esm-tools/feat/foci_mct4_smag

Feat/foci mct4 smag

seb-wahl committed:
Merge branch 'release' into geomar_dev

seb-wahl committed:
revert the foci specific change in pattern as it otherwise fails
for focioifs agrif configurations

seb-wahl committed:
added an explanation for FOCI in case restart files are not found

seb-wahl committed:
in case config["restart_in_sources"] are given explicitely
AND are not absolute paths as e.g in FOCI
ini_parent_dir: "${general.ini_parent_dir}/oasis3mct/"
    restart_in_sources: sstocean_${parent_expid}_...
we need to check for the full path as well
btw it was a nightmare to track this down

seb-wahl committed:
in case config["restart_in_sources"] are given explicitely
AND are not absolute paths as e.g in FOCI
ini_parent_dir: "${general.ini_parent_dir}/oasis3mct/"
    restart_in_sources: sstocean_${parent_expid}_...
we need to check for the full path as well
btw it was a nightmare to track this down

seb-wahl committed:
updated highresmip/focioifs-sp1950-agrif-glogin.yaml to be compatible with updates from release

joakimkjellsson committed:
focioifs 3.1 with ECWAM. Works on glogin

seb-wahl commented:
It also solves #1133

mandresm commented on configs/components/oifs/oifs.yaml:
```diff
@@ -969,7 +996,13 @@ choose_major_version:
                                         ORBIY: "${orb_iyear}"
 
                 add_outdata_files:
+                        # output is handled by XIOS
+                        # but we write the full model state at the last time step
+                        # in case we want to create restart files 
                         ICMUA: ICMUA
+                        ICMSH: ICMSH
+                        ICMGG: ICMGG 
```
@JanStreffing, are you okay with this change? Otherwise, this can be moved to `oifs.add_outdata_files` in the `focioifs.yaml`

mandresm replied:
I don't think this will be a problem, if the files are not found they should be ignored.

mandresm commented on configs/components/oifs/oifs.env.yaml:
```diff
@@ -140,7 +140,7 @@ compiletime_environment_changes:
     blogin:
       add_export_vars:
-        - "LAPACK_LIB='-mkl=sequential'"
+        - "LAPACK_LIB='-qmkl=sequential'"
```
@seb-wahl, can you explain why this flag needs the extra `q`? I'm curious to understand why is it now needed in HLRN.

mandresm commented on configs/components/oifs/oifs_grib_output_to_restart_new.sh:
```diff
@@ -0,0 +1,290 @@
+cdo -f nc -sp2gp,cubic ${outdir}/${outdate}/ICMSH${expout}INIT ${outdir}/${outdate}/ICMSH${expout}INIT.nc 
+
+
+echo " ==== Can you feel that? We are done here... "
```
XDDDD <3

joakimkjellsson committed:
rename versions of FOCI-OpenIFS. Does not work yet

seb-wahl committed:
fixed runoff coupling fields (otherwise FOCI-AGRIF will not compile

seb-wahl committed:
added MOPS via direct coupling
used for MO053D sochic extension

joakimkjellsson committed:
introduce new versions for NEMO and remove some versions. Attempt to reduce code duplication

joakimkjellsson committed:
turn of -g -traceback for OpenIFS and switch from -O1 to -O3 on glogin. Tco95 on 288 cores went from 80 min per year to 60min

joakimkjellsson committed:
remove a coupling which was badly named and not needed

joakimkjellsson committed:
add comment for sclct_switch in oifs.yaml. And set 1800s to default time step for Tco95

joakimkjellsson committed:
rename focioifs versions and add comments. Also move lots of parameters (sclct_switch, o3_scheme, coupling_time_step, etc) from runscripts to focioifs.yaml to make sure they are set by default. Still need to test all the new versions.

joakimkjellsson committed:
rename some runscripts to fit new version names

joakimkjellsson committed:
add new couplings with more reasonable names. One for 3.0 (with MCT5), one for agrif and one for non-agrif twin.

joakimkjellsson committed:
fix path for NEMO src code

joakimkjellsson committed:
runscript for FOCI-OpenIFS 3.0

joakimkjellsson committed:
rename more couplings. FOCI-OpenIFS v3.0 and v4.0 runs

joakimkjellsson committed:
I dont think the choose statement in xios.yaml is needed. It just searches for *.xml anyway

joakimkjellsson committed:
renamed runscript for FOCI-OpenIFS 4

joakimkjellsson committed:
re-configure viscosity to properly scale from ORCA05. Note: Only needs to be re-configured for each nest

joakimkjellsson committed:
renamed AGRIF config from using foci22 to foci30

joakimkjellsson committed:
use new coupling for FOCI-OpenIFS AGRIF

joakimkjellsson committed:
Merge pull request #1143 from esm-tools/focioifs_correct_versions

Correcting version names for FOCI-OpenIFS

joakimkjellsson committed:
foci-openifs 4.0 compiles on olaf. Not sure it runs

joakimkjellsson committed:
update paths on olaf

joakimkjellsson committed:
correct location for geomar libs on olaf

joakimkjellsson committed:
focioifs-agrif-4.0 compiles on glogin. Still need to add proper 1_namelist_cfg changes, input files etc

joakimkjellsson committed:
added coupling for focioifs-agrif-4.0

joakimkjellsson committed:
add input files and namelists for NEMO 4 with NWPAC10 nest

joakimkjellsson committed:
added OASIS_NETCDF and OASIS_NETCDFF for all machines (except old Mistral) to help with compiling OASIS3-MCT5 on both system libs and geomar libs

joakimkjellsson committed:
avoid duplicated add_namelist_changes in nemo.yaml. Also introduce use_si3 but use_lim2 remains default for back compatibility. Sorting out namelist changes for AGRIF in NEMO 4.2.2 but still not there I think

joakimkjellsson committed:
switch to agrif branch of runoff mapper for focioifs-agrif-4.0. Also correct names of coupling fields from AGRIF

joakimkjellsson committed:
added runscript for focioifs-agrif-4.0 on glogin

joakimkjellsson committed:
added runscript for focioifs-agrif-3.0 piControl on glogin

joakimkjellsson committed:
change from rnfmap-foci211 to rnfmap-agrif in focioifs-agrif-4.0

seb-wahl committed:
first draft including MOPS with AGRIF
esm_master get-foci-agrif_mops_oasismct4 -c
works

seb-wahl committed:
AGRIF-MOPS compiles and runs until it hits an error in trcsms_my_trc.F90
that is believed to be independent of ESM-Tools settings and rather related
to the fact that we use direct coupling / OASISMCT4

joakimkjellsson committed:
Merge pull request #1169 from esm-tools/feature/ibs_olaf

Feature/ibs olaf

joakimkjellsson committed:
added gloginp3 machine for Emmy Phase 3 (NHR)

joakimkjellsson committed:
finally got FOCI-OpenIFS-AGRIF 4.0 (NWPAC10) running on glogin (old nodes). Not happy with NEMO parameter settings, but at least it runs

joakimkjellsson committed:
added partitions etc for glogin phase 3 nodes

joakimkjellsson committed:
we always choose the old nodes on glogin. No way to figure out the difference between phase 3 and older nodes

joakimkjellsson committed:
bugfix. Set Euler step differently for NEMO 3.6 and 4.2

joakimkjellsson committed:
update for NESH (Kiel University). FOCI-OpenIFS compiles and runs.

joakimkjellsson committed:
fixes to run OpenIFS on nesh (Kiel uni)

joakimkjellsson committed:
fixes for olaf

pgierz committed:
Merge branch 'release' into geomar_dev

pgierz commented:
Merge conflicts resolved, but it looks like @joakimkjellsson is still working on this branch (recent commits in the last few days). 

I also would like one of the other main esm-tools devs to look over this, but looking at it (only briefly) I do not see any big concerns.

@seb-wahl: it would be nice if we could get a demo of the monitoring feature, I just saw the file name "notebook" somewhere while browsing, and it might be interesting for our scientists as well.

Have a nice summer holiday!

joakimkjellsson commented:
Hi @pgierz and @seb-wahl 

Yes I've been doing a few patches on `geomar_dev` to make the models run on our new machines. 

If needed, I can pause for a while to make the merging easier. 

Cheers
Joakim

pgierz commented:
Hey @joakimkjellsson,

No problem, keep hacking away ;-)
I'll occasionally merge release into this one, so you might want to a `git pull` once in a while.

MfG PG

joakimkjellsson commented:
Unfortunately, something must have gone wrong during the merge. I did `git pull` just now and nothing works. 
For example
```bash
(base) [smomw352@nesh-login1 models_oneapi]$ esm_master comp-focioifs-agrif-4.0 -c
(base) [smomw352@nesh-login1 models_oneapi]$
```
i.e. `esm_master` does not do anything. No error but no result. 

Any ideas? Any big changes that happened? 

/Joakim

pgierz commented:
Hmm, that's odd...sorry about that, I'll have a look once I'm in the office. For now you can just revert the merge so you can keep working. For that, you can do something like this:

```
$ git rebase -i HEAD^^ # Set's your git back by two commits, and opens up the commits in an editor
... edit away the merge commit
... close editor
$ git commit -m "removes merge"
$ git push     
```

I'm not 100% sure, and the internet is dropping in and out on my train, but that should put you back into a workable state.

pgierz committed:
fix(esm_parser.py): no more complex numbers

seb-wahl committed:
preliminary changes made to get FOCI work on blogin
running rocky linux 9
OASISMCT5 compiles but echam6 and xios r1910 still have issues

Commit:
Upgrade oifs-43r3-v2 to use OASIS-MCT5 and XIOS trunk since older versions struggle to compile on Intel 2023. Should not influence results in any way

Commit:
update environment changes for OpenIFS compilation on blogin with Intel 2024. Some flags etc were no longer supported by the Intel compilers and some behind-the-scenes defaults have changed

Commit:
set new default compilers for Intel

joakimkjellsson committed:
my poor attempt to run on glogin. Added OpenMPI option for NEMO

joakimkjellsson committed:
added new options for OpenIFS, XIOS and rnfmap on glogin

joakimkjellsson committed:
remove gloginp3. We will just have glogin

Commit:
fixes to compile FOCI-OpenIFS on blogin. Must change in arch files to use cpp rather than mpiicc as preprocessor

joakimkjellsson committed:
focioifs now compiles with both intel+intelmpi and intel+openmpi. but does not run with either... :-(

joakimkjellsson committed:
trying to use intel+openmpi on glogin. Crashed with srun. Hangs with mpirun

joakimkjellsson committed:
update runscript for focioifs with intel+openmpi on glogin

joakimkjellsson committed:
new compiler_mpi=intel2021_impi2019 using spack on glogin. Still FOCI-OpenIFS does not work. Likely that problem is with the system config and not the model or Intel MPI

joakimkjellsson committed:
SUCCESS! FOCI-OpenIFS 3.0 now runs on glogin. Still unclear about FOCI-OpenIFS 4.0 or FOCI. Had to downgrade netcdf modules (see comment in glogin.yaml) and switch from srun to mpirun. It seems that NEMO 4.2 requires xios_trunk but xios_trunk hangs at the last time step. So FOCI-OpenIFS 4.0 does not run. Will need more investigation.

joakimkjellsson committed:
fix path to NCO and CDO on glogin

joakimkjellsson committed:
modify runscripts for FOCI-OpenIFS 4.0 and 3.0-AGRIF for new glogin

joakimkjellsson committed:
Merge branch 'geomar_dev' into feature/blogin-rockylinux9

joakimkjellsson committed:
Merge pull request #1210 from esm-tools/feature/blogin-rockylinux9

Feature/blogin rockylinux9

joakimkjellsson committed:
update oifs env for new nesh, and remove mistral settings

joakimkjellsson committed:
switch to new compilers, icc->icx, ifort->ifx

joakimkjellsson committed:
set default nest1: false, instead of no. focioifs.yaml needs to evaluate nest1 as true or false

joakimkjellsson committed:
add restart file to restart Tco399+ORCA12 on glogin new nodes

Commit:
added for echam export vars for oasis
changed for nemo from shlat 2 -> 0
nn_closea 1 -> 0
fixed bgc passive tracer definitions
changed the coupling definitions -> oasis version 5.0, xios2.5 r1910
new compiler settings for intel2023_impi2021 (glogin specific with intel_oneapi, maybe this has to be done cleaner?)
increased debug level
some changes to the runscript yaml

joakimkjellsson committed:
fix small syntax error for nesh.yaml

joakimkjellsson committed:
Merge branch 'geomar_dev' of https://github.com/esm-tools/esm_tools into geomar_dev

Commit:
added missing IDEAL tracer definition for FOCI_AGRIF_MOPS configuration

Commit:
Added missing configuration for foci_mops

Commit:
Adapted runscripts to a pre final version -> further test necessary

seb-wahl committed:
Merge remote-tracking branch 'origin/geomar_dev' into geomar

seb-wahl committed:
Merge remote-tracking branch 'origin/feature/mops_agrif_direct_coupling' into geomar_dev
merge Tronje's MOPS developments into geomar_dev

seb-wahl committed:
make FOCI-MOPS work again after the merge of Tronjes and
Joakims developments, remove a few hardcoded paths

joakimkjellsson committed:
added some comments on namelist changes for nemo in FOCI-OpenIFS 4.0

joakimkjellsson committed:
Ladies and Gentlemen! Introducing: FOCI-OpenIFS 4.1 with eORCA05.L75 grid

seb-wahl committed:
Merge branch 'release' into feature/merge_release_geomar_dev

seb-wahl committed:
Merge remote-tracking branch 'origin/release' into feature/merge_release_geomar_dev

Commit:
Added a yaml file to restart ESM simulations with ssp scenarios from 2014 with FOCI-MOPS
and added bugfixes to the config files.

seb-wahl committed:
Merge branch 'geomar_dev' of https://github.com/esm-tools/esm_tools into geomar_dev

seb-wahl committed:
Merge branch 'geomar_dev' into geomar

seb-wahl committed:
fixed name pattern for glogin standard96 compute nodes

seb-wahl committed:
Merge branch 'geomar_dev' into geomar

TronjeKemena committed:
Update oasis.py

Bugfix bracket not closed

Commit:
Added git to the modules. This is no more a standard librabry since the update.

seb-wahl committed:
load git-lfs by default as no system git is available anymore

seb-wahl committed:
fix a merge error in geomar branch

seb-wahl committed:
fixed echam monitoring, added comment to NEMO monitoring

seb-wahl committed:
added perl modules for XIOS, NEMO and OIFS compilation, replaces custom perl installation

seb-wahl committed:
added draft for FOCIOIFS-4.0 runscript on levante for Rene

seb-wahl committed:
fixed coupling for OASISMCT4 with foci-default
(need to overwrite FOCOIFS coupling defaults to be able to
use the same NEMO config's as FOCIOIFS)

seb-wahl committed:
Merge branch 'release' into geomar_dev

mandresm commented on configs/components/oasis3mct/oasis3mct.yaml:
```diff
@@ -4,7 +4,7 @@ cf_name_dir: ${model_dir}
 time_step: 900 # changed by coupled setup anyway
 norestart: F
 use_lucia: False
-mct_version: "2.8" # TODO: set mct_version = version (can't do at the moment as version is a string)
```
does this change affect AWIESM2/AWICM3?

mandresm replied:
It was affecting but I pushed a commit to fix that. Now `mct_version: 2.8` is set in the setup files of `awicm-1` and `awiesm-2`.

mandresm commented on configs/components/oifs/oifs.env.yaml:
```diff
@@ -27,7 +25,7 @@ runtime_environment_changes:
 
     glogin:
-      add_module_actions:
+      add_module_actions: 
```
```suggestion
      add_module_actions:
```

mandresm commented on configs/components/oifs/oifs.yaml:
```diff
@@ -347,7 +393,12 @@ choose_resolution:
+                # This has 2x computer cost but reduces
+                # biases by a lot, so it is worth it
+                # Savita et al. (2024) 10.5194/gmd-2023-101 
+                #time_step: 3600
```
```suggestion
```

mandresm commented on configs/components/oifs/oifs.yaml:
```diff
@@ -1297,18 +1375,63 @@ choose_output:
 # as general.run_number, but for branchoff experiments should be 0 so that the 3rd
 # preprocessing option is used, instead of the options 1 or 2.
 branchoff: false
+ini_pseudo_initial_date: "${initial_date}" #${prev_run.general.start_date}"  
```
I don't think this can go in. It will break things pretty hard, but I still need to check lines below

mandresm replied:
Indeed, this can go in :) Now I get it

mandresm commented on configs/components/oifs/oifs.yaml:
```diff
@@ -1297,18 +1375,63 @@ choose_output:
-    ))"
+    ))" 
+
+# There are four kinds of runs:
+# eternal_run_number=0 - Restart from another experiment. 
+# eternal_run_number=1 - Start from initial conditions
+# eternal_run_number=2 - Regular restart
+# eternal_run_number=3 - Also regular restart, but we trick OpenIFS to think 
+#                        the initial_date is the start_date of the previous run
+#                        i.e. OpenIFS never knows it does more than 1 restart
+#                        This avoids both a memory issue for long runs and 
+#                        overflow of some calendar variables in single precision. 
+#
+# When we branch off from an existing experiment, we must set ini_pseudo_initial_date
+# in the runscript to inform ESM-Tools what initial_date it should trick OpenIFS it is.
+# For example, a piControl may start from 1850 and run to 2850. 
+# A historical run would then start from 2850 of the existing run, but we would set 
+# start_date for the new run to be 1850 and also trick OpenIFS to think initial_date is 1849. 
```
Amazing comment, thank you so much @joakimkjellsson!!!

mandresm commented on configs/components/oifs/oifs.yaml:
```diff
@@ -1357,9 +1482,9 @@ choose_eternal_run_number:
-                                branchoff: true
+                #choose_eternal_run_number:
+                #        0:
+                #                branchoff: true
```
```suggestion
```

mandresm commented on configs/components/oifs/oifs.yaml:
```diff
@@ -1388,11 +1513,30 @@ preprocess:
+#postprocess:
+#        postprocess_shell:
+#                method: "${general.esm_function_dir}/components/oifs/oifs-43r3-postprocess.sh ${work_dir} ECE3 ${start_date!syear!smonth!sday} ${end_date!syear!smonth!sday}"
+#                type: shell
```
```suggestion
```

mandresm commented on configs/machines/all_machines.yaml:
```diff
@@ -37,10 +37,17 @@ blogin:
+#        login_nodes: 'glogin1[1-3]'
+#        compute_nodes: '^g[A-Za-z0-9]+\.usr\.hlrn\.de$'
+#        post_nodes: 'glogin1[1-3]'
+
```
```suggestion
```

mandresm commented on src/esm_environment/esm_environment.py:
```diff
@@ -548,8 +548,11 @@ def write_dummy_script(self, include_set_e=True):
+            #)
             script_file.write(
-                f'#!{self.config.get("sh_interpreter", "/bin/bash")} -l\n'
+                f'#!{self.config.get("sh_interpreter", "/bin/bash")}\n'
```
I cannot recall well why, but this breaks stuff in other machines. Suggest to remove it from here as you have it done currently and put it back in the other machines you don't use under `sh_interpreter`. I'll take care of that.

mandresm commented on src/esm_environment/esm_environment.py:
```diff
@@ -548,8 +548,11 @@ def write_dummy_script(self, include_set_e=True):
             print('WARNING: "sh_interpreter" not defined in the machine yaml')
         with open("dummy_script.sh", "w") as script_file:
             # Write the file headings
+            #script_file.write(
+            #    f'#!{self.config.get("sh_interpreter", "/bin/bash")} -l\n'
+            #)
```
```suggestion
```

mandresm commented on src/esm_runscripts/oasis.py:
```diff
@@ -46,8 +46,11 @@ def __init__(
-            if mct_version >= (5, 0):
-                self.namcouple += [" $NLOGPRT", "           " + str(debug_level) + " 0 1", " $END"]
+            # LUCIA (load balancing) is done differently in MCT 5.0
+            if mct_version >= (5,0):
```
```suggestion
            if mct_version >= (5, 0):
```

mandresm commented on src/esm_runscripts/oasis.py:
```diff
@@ -46,8 +46,11 @@ def __init__(
+            if mct_version >= (5,0):
+                # In MCT5 you set X Y Z, where X refers to verbosity, Y to timing info and Z to load balancing
+                # Here: Set X = debug_level, Y = 0 (no info), Z = 1 (activate load balancing)
+                self.namcouple += [" $NLOGPRT", "           " + str(debug_level) + " 0 1 ", " $END"]
```
```suggestion
                self.namcouple += [" $NLOGPRT", "           " + str(debug_level) + " 0 1", " $END"]
```

mandresm commented on src/esm_runscripts/oasis.py:
```diff
@@ -541,12 +544,9 @@ def add_restart_files(self, restart_file_label, fconfig):
-        ):
+        # Not the rstas.nc soft link to the last, but the actual one for the
+        # branch-off date
+        if gconfig["run_number"] == 1 and config["lresume"] and gconfig["jobtype"] == "prepcompute" and config.get("norestart", "F") == "F":
```
```suggestion
        if (
            gconfig["run_number"] == 1
            and config["lresume"]
            and gconfig["jobtype"] == "prepcompute"
            and config.get("norestart", "F") == "F"
        ):
```

mandresm reviewed (changes requested):
It all looks great and not problematic. The changes in the `src` need though to be tested more extensively. Consequently, what I will do will be to create a new branch from this one, and merge almost everything but the changes in `src`. The changes in `src` can be then tested more throughly in this PR.

mandresm committed:
Update configs/components/oifs/oifs.env.yaml

mandresm committed:
Update configs/components/oifs/oifs.yaml

mandresm committed:
Update configs/components/oifs/oifs.yaml

mandresm committed:
Update configs/components/oifs/oifs.yaml

mandresm committed:
Update configs/machines/all_machines.yaml

mandresm committed:
add mct_version: 2.8 to all the setups that were depending on mct 2.8

mandresm commented:
Prior to merging "updating" this with `release` or merging this into `release` make sure to merge this #1299
